x-pack/plugins/security_solution/server/lists_integration/endpoint/validators/blocklist_validator.ts

```diff
@@ -0,0 +1,284 @@
+ * 2.0.
+ */
+
+import { ENDPOINT_BLOCKLISTS_LIST_ID } from '@kbn/securitysolution-list-constants';
+import { schema, TypeOf } from '@kbn/config-schema';
+import { ExceptionListItemSchema } from '@kbn/securitysolution-io-ts-list-types';
+import { OperatingSystem } from '@kbn/securitysolution-utils';
+import { BaseValidator } from './base_validator';
+import { ExceptionItemLikeOptions } from '../types';
+import {
+  CreateExceptionListItemOptions,
+  UpdateExceptionListItemOptions,
+} from '../../../../../lists/server';
+import { isValidHash } from '../../../../common/endpoint/service/trusted_apps/validations';
+import { EndpointArtifactExceptionValidationError } from './errors';
+
+const ProcessHashField = schema.oneOf([
+  schema.literal('process.hash.md5'),
+  schema.literal('process.hash.sha1'),
+  schema.literal('process.hash.sha256'),
+]);
+const ProcessExecutablePath = schema.literal('process.executable.caseless');
+const ProcessCodeSigner = schema.literal('process.Ext.code_signature');
+
+const ConditionEntryTypeSchema = schema.literal('match_any');
+const ConditionEntryOperatorSchema = schema.literal('included');
+
+type ConditionEntryFieldAllowedType =
+  | TypeOf<typeof ProcessHashField>
+  | TypeOf<typeof ProcessExecutablePath>
+  | TypeOf<typeof ProcessCodeSigner>;
+
+type BlocklistConditionEntry =
+  | {
+      field: ConditionEntryFieldAllowedType;
+      type: 'match_any';
+      operator: 'included';
+      value: string[];
+    }
+  | TypeOf<typeof WindowsSignerEntrySchema>;
+
+/*
+ * A generic Entry schema to be used for a specific entry schema depending on the OS
+ */
+const CommonEntrySchema = {
+  field: schema.oneOf([ProcessHashField, ProcessExecutablePath]),
+  type: ConditionEntryTypeSchema,
+  operator: ConditionEntryOperatorSchema,
+  // If field === HASH then validate hash with custom method, else validate string with minLength = 1
+  value: schema.conditional(
+    schema.siblingRef('field'),
+    ProcessHashField,
+    schema.arrayOf(
+      schema.string({
+        validate: (hash: string) =>
+          isValidHash(hash) ? undefined : `invalid hash value [${hash}]`,
+      })
+    ),
+    schema.conditional(
+      schema.siblingRef('field'),
+      ProcessExecutablePath,
+      schema.arrayOf(
+        schema.string({
+          validate: (pathValue: string) =>
+            pathValue.length > 0 ? undefined : `invalid path value [${pathValue}]`,
+        })
+      ),
+      schema.arrayOf(
+        schema.string({
+          validate: (signerValue: string) =>
+            signerValue.length > 0 ? undefined : `invalid signer value [${signerValue}]`,
+        })
+      )
+    )
+  ),
+};
+
+// Windows Signer entries use a Nested field that checks to ensure
+// that the certificate is trusted
+const WindowsSignerEntrySchema = schema.object({
+  type: schema.literal('nested'),
+  field: ProcessCodeSigner,
+  entries: schema.arrayOf(
+    schema.oneOf([
```

Comment:
Lets check on this one too. I though that I saw some thread that said for blocklists, the cert does not have to be signed.

cc/ @joeypoon ☝️